File: API/App.Model/Entities/CoachCard.cs
using App.Model.Entities;
using App.Model.Entities.Common;
using App.Model.Enums;
using System.ComponentModel.DataAnnotations;

namespace App.Model.Entities
{
    public class CoachCard : AuditableEntity
    {
        public Coach Coach { get; set; }
        public Match Match { get; set; }

        [Required]
        public CardColor Color { get; set; }
        [Required]
        public int Count { get; set; }
    }
}

File: WebUI/app-frontend/src/app/models/cards/coach-card.ts
import { CardColor } from "src/app/enums/card-color";
import { SimpleCoach } from "../coach/simple-coach.model";

export class CoachCard {
    public id : string;
    public isActive : boolean;
    public coach : SimpleCoach;
    public matchId : string;
    public color : CardColor;
    public count : number;
}
